Region/BasicStatsCalculator.tcc

```diff
@@ -0,0 +1,100 @@
+#ifndef CARTA_BACKEND_REGION_BASICSTATSCALCULATOR_TCC_
+#define CARTA_BACKEND_REGION_BASICSTATSCALCULATOR_TCC_
+
+#include "BasicStatsCalculator.h"
+
+namespace carta {
+
+template<typename T>
+void BasicStats<T>::join(BasicStats<T>& other) {
+    if (other.num_pixels) {
+        sum += other.sum;
+        sumSq += other.sumSq;
+        num_pixels += other.num_pixels;
+        min_val = std::min(min_val, other.min_val);
+        max_val = std::max(max_val, other.max_val);
+        mean = sum / num_pixels;
+        stdDev = sqrt(sumSq / num_pixels - mean * mean);
```

Comment:
In the HDF5 loader calculation for the standard deviation I use a slightly different formula for consistency with casacore: `sqrt((sum_sq - (sum * sum / num_pixels)) / (num_pixels - 1))`, rather than `sqrt((sum_sq - (sum * sum / num_pixels)) / num_pixels)`, which is equivalent to your formula. The relevant function in casacore is `LattStatsSpecialize::getSigma`, which calls `LattStatsSpecialize::getVariance`. Elsewhere in casacore your formula is used, but I'm pretty sure that the variant is always used for lattice regions, and that I used it to make the results match casacore's.